scripts/dts/edtlib.py

```diff
@@ -2085,6 +2540,33 @@ def _phandle_val_list(prop, n_cells_name):
     return res
 
 
+def _partitions_list(partitions_node, flash_node):
+    # Returns a list of ControllerAndData objects for partitions
+
+    partitions_list = []
+    for partition_name, partition_node in partitions_node.children.items():
+        partition = ControllerAndData()
+        partition.node = partition_node
+        partition.controller = flash_node
+        partition.data = {}
+        for prop_name, prop in partition_node.props.items():
+            partition.data[prop_name] = prop.val
+        if 'partitions' in partition_node.children:
+            child_partitions = partition_node.children['partitions']
+            partition.data['partitions'] = _partitions_list(child_partitions,
```

Comment:
I think we should avoid magic hardcoded strings like these that "piggyback" on dictionaries used for other stuff. Better to use regular attributes.

Maybe a new class could be added.

Reply:
New class 'Partition' used.